Mathlib/Topology/Algebra/Valued/ValuativeRel.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.RingTheory.Valuation.ValuativeRel
+import Mathlib.Topology.Algebra.Valued.ValuationTopology
+
+/-!
+
+# Valuative Relations as Valued
+
+In this temporary file, we provide a helper instance
+for `Valued R Γ` derived from a `ValuativeRel R`,
+so that downstream files can refer to `ValuativeRel R`,
+to facilitate a refactor.
+
+-/
+
+namespace ValuativeRel
+
+variable {R : Type*} [CommRing R]
+
+instance [UniformSpace R] [IsUniformAddGroup R] [ValuativeRel R] [ValuativeTopology R] :
+    Valued R (ValueGroupWithZero R) :=
+  .mk (valuation R) ValuativeTopology.mem_nhds_iff
+
+end ValuativeRel
+
+namespace ValuativeTopology
+
+variable {R : Type*} [CommRing R] [ValuativeRel R] [TopologicalSpace R] [ValuativeTopology R]
+
+open ValuativeRel TopologicalSpace Filter Topology Set
+
+local notation "v" => valuation R
+
+variable (R) in
+theorem hasBasis_nhds_zero :
+    (𝓝 (0 : R)).HasBasis (fun _ => True)
+      fun γ : (ValueGroupWithZero R)ˣ => { x | v x < γ } := by
+  simp [Filter.hasBasis_iff, mem_nhds_iff]
+
+variable [IsTopologicalAddGroup R]
+
+theorem mem_nhds {s : Set R} {x : R} :
+    s ∈ 𝓝 x ↔ ∃ γ : (ValueGroupWithZero R)ˣ, { y | v (y - x) < γ } ⊆ s := by
+  simp only [← nhds_translation_add_neg x, ← sub_eq_add_neg, preimage_setOf_eq, true_and,
+    ((hasBasis_nhds_zero R).comap fun y => y - x).mem_iff]
+
+theorem isOpen_ball (r : ValueGroupWithZero R) :
+    IsOpen {x | v x < r} := by
+  rw [isOpen_iff_mem_nhds]
+  rcases eq_or_ne r 0 with rfl|hr
+  · simp
+  intro x hx
```

Comment:
```suggestion
  rcases eq_or_ne r 0 with rfl | hr
  · simp
  · intro x hx
```
or
```
  rcases eq_or_ne r 0 with rfl | hr; simp
  intro x hx
```
although I think I prefer the first. These suggestions come from my reading of the style guide, which says "but even in these scenarios, newlines are preferred" regarding my two suggestions.